tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,298 @@
+
+			logrus.Infof("cluster scan is %s", clusterScanStatusType.Display.State)
+			return false, nil
+		}
+
+		return true, nil
+	})
+	require.NoError(c.T(), err)
+
+	c.CheckScanReport(steveClient, scanName)
+}
+
+func (c *CisBenchmarkTestSuite) CheckScanReport(steveClient *v1.Client, scanName string) {
+	scanReportList, err := steveClient.SteveType(charts.ClusterScanReportType).List(nil)
+	require.NoError(c.T(), err)
+
+	scanReportIdx := -1
+
+	for idx, scanReport := range scanReportList.Data {
+		if strings.Contains(scanReport.Name, scanName) {
+			scanReportIdx = idx
+			break
+		}
+	}
+
+	require.GreaterOrEqualf(c.T(), scanReportIdx, 0, "scan report not found")
+	cisReportSpec := &charts.ClusterScanReportSpec{}
+	scanReportSpec := scanReportList.Data[scanReportIdx].Spec
+	err = v1.ConvertToK8sType(scanReportSpec, cisReportSpec)
+	require.NoError(c.T(), err)
+
+	reportData := &charts.CisReport{}
+	err = json.Unmarshal([]byte(cisReportSpec.ReportJson), &reportData)
+	require.NoError(c.T(), err)
+	logrus.Infof("Out of total number of %d scans, %d scans passed, %d scans were skipped and %d scans failed", reportData.Total, reportData.Pass, reportData.Skip, reportData.Fail)
+	for _, group := range reportData.Results {
+		for _, check := range group.Checks {
+			assert.NotEqualf(c.T(), "fail", check.State, "check failed: id: %s state: %s description: %s ", check.Id, check.State, check.Description)
+		}
+	}
+
+	require.Equalf(c.T(), reportData.Fail, 0, "cluster scan failed")
+}
+
+func (c *CisBenchmarkTestSuite) TestUpgradeCisbenchmarkChart() {
+	subSession := c.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := c.client.WithSession(subSession)
+	require.NoError(c.T(), err)
+
+	// Change cisbenchmark install option version to previous version of the latest version
+	versionsList, err := client.Catalog.GetListChartVersions(charts.RancherCisBenchmarkName, catalog.RancherChartRepo)
+	require.NoError(c.T(), err)
+
+	if len(versionsList) < 2 {
+		c.T().Skip("Skipping the upgrade case, only one version of cis-benchmark is available")
+	}
+	assert.GreaterOrEqualf(c.T(), len(versionsList), 2, "There should be at least 2 versions of the cis-benchmark chart")
+	versionLatest := versionsList[0]
+	c.T().Log(versionLatest)
+
+	versionBeforeLatest := versionsList[1]
+	c.T().Log(versionBeforeLatest)
+	c.chartInstallOptions.Version = versionBeforeLatest
+
+	c.T().Log("Checking if the cis-benchmark chart is installed with one of the previous versions")
+	initialCisbenchmarkChart, err := charts.GetChartStatus(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, charts.RancherCisBenchmarkName)
+	require.NoError(c.T(), err)
+
+	if initialCisbenchmarkChart.IsAlreadyInstalled && initialCisbenchmarkChart.ChartDetails.Spec.Chart.Metadata.Version == versionLatest {
+		c.T().Skip("Skipping the upgrade case, cis-benchmark chart is already installed with the latest version")
+	}
+
+	if !initialCisbenchmarkChart.IsAlreadyInstalled {
```

Comment:
it seems like you want this to have the ability to upgrade a cluster's CIS chart if it is already installed but not on the latest version. If that's the case, wouldn't you want the previous comment's assert to be after the check if CIS is already installed? use case: cis is installed on an older / not listed version, and chartRepo from rancher only shows 1 version of CIS available. Is that valid and worth switching the order or improving the assert case in the previous comment?

Reply:
No, the upgrade test scenario will be skipped if the CIS Benchmark chart is already installed on the cluster. When the chart is not installed on a cluster, the automation will install a chart, one version previous to latest then upgrade it to latest.